### ISSUE TITLE
Support python3

### ISSUE THREAD

**Issue by ry** [denoland/deno#2893]
We are heavily dependent on V8, which in turn is heavily dependent on Chromium tooling, which does not support Python3 and has only recently begun conversion -
https://bugs.chromium.org/p/chromium/issues/detail?id=942720
https://bugs.chromium.org/p/v8/issues/detail?id=8594
https://github.com/nodejs/node/issues/24512

**Comment by hayd** [comment]
I'm happy to do this, I think it'll be straightforward, but it's probably worth waiting for v8 support to be completed.

**Comment by cclauss** [comment]
> it's probably worth waiting

Not!  Python 3 was released 11 years ago and there are 113 days until Python 2 end of life.  Please do work in _parallel_, not serial to be ready before yearend.  It would be great if someone could get v8/v8#33 and v8/v8#34 pushed and landed upstream.

**Comment by hayd** [comment]
@cclauss I bet I can do the deno part of the port in under an hour, certainly less than a day. I've ported large modules to python 3... deno is very small. I suppose it would help to enumerate precisely which python v8 scripts need to be ported in order for deno to work, I can do that.

**Comment by cclauss** [comment]
Not a question of your skills.  I have been impressed.  It is just a question of putting lots of concentrated pressure on a few bottlenecks that are not ready so we create a _compelling reason to act_.  We want to create the impression that the train is leaving the station without the laggards onboard.

**Comment by kt3k** [comment]
https://travis-ci.com/bmsdave/v8/builds/132539826
Does this mean v8 can be built with python 3?

**Comment by cclauss** [comment]
@bmsdave Can V8 be built with Python 3?

**Comment by hayd** [comment]
No: https://github.com/bmsdave/v8/commit/845d137e411fb86a06b7019f590dfd806b36dc44

**Comment by cclauss** [comment]
https://travis-ci.com/bmsdave/v8/builds Seems to be passing the lint steps.  Where does the V8 build on Py3 break down?

**Comment by hayd** [comment]
The script it's running is `true`, the Travis build is not running any build/tests other than flake8...

**Comment by cclauss** [comment]
Exactly.  So where can we see the __build__ failing on Python 3?

**Comment by hayd** [comment]
Sorry I was answering @kt3k :

> https://travis-ci.com/bmsdave/v8/builds/132539826
> Does this mean v8 can be built with python 3?

It could be that the build works with just the syntax changes (but I would be surprised).

I think we need to compile and run with [gm.py](https://github.com/v8/v8/blob/master/tools/dev/gm.py), see https://v8.dev/docs/test.

**Comment by kitsonk** [comment]
Closing in favour of #8223 (plus we have #7779).
